wzdx/experimental_combination/attenuator.py

```diff
@@ -0,0 +1,285 @@
+        wzdx[0]["features"][0]["properties"]["start_date"] = (
+            date_tools.get_iso_string_from_datetime(datetime.now() - timedelta(days=1))
+        )
+        wzdx[0]["features"][0]["properties"]["end_date"] = (
+            date_tools.get_iso_string_from_datetime(datetime.now() + timedelta(days=1))
+        )
+
+    combined_events = get_combined_events(geotab_avl, wzdx)
+
+    with open(outputPath, "w+") as f:
+        f.write(json.dumps(combined_events, indent=2))
+        logging.debug(f"Output written to {outputPath}")
+
+
+def validate_directionality(geotab, wzdx):
+    geotab_bearing = geotab["avl_location"]["position"]["bearing"]
+    wzdx_direction = wzdx["features"][0]["properties"]["core_details"]["direction"]
+
+    geotab_direction = geospatial_tools.get_closest_direction_from_bearing(
+        geotab_bearing, wzdx_direction
+    )
+
+    return geotab_direction == wzdx_direction
+
+
+def validate_dates(geotab, wzdx):
+    geotab_date = date_tools.get_unix_from_iso_string(
+        geotab["avl_location"]["source"]["collection_timestamp"]
+    )
+    logging.debug(f"Geotab: {json.dumps(geotab)}")
+    if not geotab_date:
+        geotab_date = (
+            geotab["avl_location"]["source"]["collection_timestamp"] * 1000
+            if geotab["avl_location"]["source"]["collection_timestamp"]
+            else None
+        )
+    if not geotab_date:
+        logging.debug(f"No geotab date found")
+        return False
+
+    wzdx_start_date = date_tools.get_unix_from_iso_string(
+        wzdx["features"][0]["properties"]["start_date"]
+    )
+    wzdx_end_date = date_tools.get_unix_from_iso_string(
+        wzdx["features"][0]["properties"]["end_date"]
+    )
+
+    logging.debug(f"Dates: {wzdx_start_date}, {geotab_date}, {wzdx_end_date}")
+
+    return wzdx_start_date <= geotab_date <= wzdx_end_date
+
+
+def get_combined_events(geotab_msgs, wzdx_msgs):
+    active_wzdx_msgs = wzdx_translator.filter_active_wzdx(wzdx_msgs)
+
+    combined_events = []
+    for i in identify_overlapping_features(geotab_msgs, active_wzdx_msgs):
+        geotab_msg, wzdx_msg = i
+        event_status = wzdx_translator.get_event_status(wzdx_msg["features"][0])
+        if event_status in ["active"]:
+            wzdx = combine_geotab_with_wzdx(geotab_msg, wzdx_msg)
+            combined_events.append(wzdx)
+    return combined_events
+
+
+def identify_overlapping_features(geotab_msgs, wzdx_msgs):
+    geotab_routes = {}
+    matching_routes = []
+
+    for geotab_msg in geotab_msgs:
+        geometry = geotab_msg["avl_location"]["position"]
+
+        route_details = cdot_geospatial_api.GeospatialApi().get_route_and_measure(
+            (geometry["latitude"], geometry["longitude"])
+        )
+        if not route_details:
+            continue
+        geotab_msg["route_details_start"] = route_details
+        geotab_msg["route_details_end"] = None
+        logging.debug(route_details)
+
+        if not route_details:
```

Comment:
This code shouldn't ever get hit since we `continue` just above this for the same condition. Should we pull the logging statement up to that line?